src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -626,5 +681,29 @@ private BitbucketRepository getRepositoryByNameOrSlug(String projectNameOrKey, S
                                     .getRepositoryClient(repositoryNameOrSlug)
                                     .getRepository()));
         }
+
+        private BitbucketRepository getRepository(BitbucketServerConfiguration server, String projectKey,
+                                                  String repositorySlug, @Nullable String credentialsId) {
+            BitbucketClientFactory client = bitbucketClientFactoryProvider
+                    .getClient(server.getBaseUrl(), jenkinsToBitbucketCredentials.toBitbucketCredentials(credentialsId, server));
+            return getRepository(projectKey, repositorySlug, client);
+        }
+
+        private BitbucketRepository getRepository(String projectKey, String repoSlug,
+                                                  BitbucketClientFactory clientFactory) {
+            return clientFactory.getProjectClient(projectKey)
+                    .getRepositoryClient(repoSlug)
+                    .getRepository();
+        }
+
+        private BitbucketMirrorHandler createMirrorHandlerUsingRepoSearch() {
+            return new BitbucketMirrorHandler(bitbucketPluginConfiguration, bitbucketClientFactoryProvider, jenkinsToBitbucketCredentials,
+                    (client, bbRepo) -> this.getRepositoryByNameOrSlug(bbRepo.getProjectNameOrKey(), bbRepo.getRepoNameOrSlug(), client));
```

Comment:
/nit unnecessary `this`

Reply:
This is now changed to the version which is based on your branch. `this` is not there anymore